src/app/products/product-details/product-details.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit } from '@angular/core';
+import { Input } from '@angular/core';
+import { Product } from 'src/app/models/Product';
+import { ProductService } from '../../core/ProductService';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.scss'],
+})
+export class ProductDetailsComponent implements OnInit {
+  @Input()
+  product: Product;
+  subscription: Subscription;
```

Comment:
Widzę że jest przypisanie do tego ale na tym się kończy.

getProductById zwraca observable który jest "completed on arrival", right? więc to nie jest w ogóle potrzebne.

Reply:
ok